dispatches/models/fossil_case/properties/hitecsalt_properties.py

```diff
@@ -0,0 +1,424 @@
+        if state_vars_fixed is False:
+            # Fix state variables if not already fixed
+            Fflag = {}
+            Tflag = {}
+            Pflag = {}
+            
+            for k in blk.keys():
+                if blk[k].flow_mass.fixed is True:
+                    Fflag[k] = True
+                else:
+                    Fflag[k] = False
+                    if state_args is None:
+                        blk[k].flow_mass.fix()
+                    else:
+                        blk[k].flow_mass.fix(state_args["flow_mass"])
+
+                if blk[k].pressure.fixed is True:
+                    Pflag[k] = True
+                else:
+                    Pflag[k] = False
+                    if state_args is None:
+                        blk[k].pressure.fix()
+                    else:
+                        blk[k].pressure.fix(state_args["pressure"])
+
+                if blk[k].temperature.fixed is True:
+                    Tflag[k] = True
+                else:
+                    Tflag[k] = False
+                    if state_args is None:
+                        blk[k].temperature.fix()
+                    else:
+                        blk[k].temperature.fix(state_args["temperature"])
+
+            flags = {"Fflag": Fflag, "Tflag": Tflag, "Pflag": Pflag}
+
+        else:
+            # Check when the state vars are fixed already result in dof 0
+            for k in blk.keys():
+                if degrees_of_freedom(blk[k]) != 0:
+                    raise Exception("State vars fixed but degrees of freedom "
+                                    "for state block is not zero during "
+                                    "initialization.")
+
+        # Set solver options
+        if outlvl > 1:
+            stee = True
+        else:
+            stee = False
+
+        opt = get_solver(solver, optarg)
+
+        # ---------------------------------------------------------------------
+        # Solve property correlation
+        results = solve_indexed_blocks(opt, [blk], tee=stee)
+
+        if outlvl > 0:
+            if results.solver.termination_condition \
+                    == TerminationCondition.optimal:
+                _log.info('{} Initialisation Step 1 Complete.'
+                          .format(blk.name))
+            else:
+                _log.warning('{} Initialisation Step 1 Failed.'
+                             .format(blk.name))
+
+        # ---------------------------------------------------------------------
+        if state_vars_fixed is False:
+            # release state vars fixed during initialization if control
+            # volume didn't fix the state vars
+            if hold_state is True:
+                return flags
+            else:
+                blk.release_state(flags)
+
+        if outlvl > 0:
+            if outlvl > 0:
+                _log.info('{} Initialisation Complete.'.format(blk.name))
+
+    def release_state(blk, flags, outlvl=0):
```

Comment:
You should use the `revert_state_vars` utility method for this.

Reply:
addressed